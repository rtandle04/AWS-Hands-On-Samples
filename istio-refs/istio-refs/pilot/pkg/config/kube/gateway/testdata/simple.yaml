apiVersion: networking.x.k8s.io/v1alpha1
kind: GatewayClass
metadata:
  name: istio
spec:
  controller: istio.io/gateway-controller
---
apiVersion: networking.x.k8s.io/v1alpha1
kind: Gateway
metadata:
  name: gateway
  namespace: istio-system
spec:
  class: gatewayclass
  listeners:
  - name: primary
    address:
      type: NamedAddress
      value: my.domain.example
    port: 80
    protocol: http
  routes:
  - group: networking.x-k8s.io/v1alpha1
    resource: HTTPRoute
    name: http
  - group: networking.x-k8s.io/v1alpha1
    resource: TcpRoute
    name: tcp
---
apiVersion: networking.x.k8s.io/v1alpha1
kind: HTTPRoute
metadata:
  name: http
  # TODO use different namespace
  namespace: istio-system
spec:
  hosts:
  - hostname: "my.domain.example"
    rules:
    - match:
        pathType: Prefix
        path: /get
        header:
          my-header: some-value
      filter:
        headers:
          add:
            my-added-header: added-value
          remove: [my-removed-header]
      action:
        forwardTo:
          group: v1
          resource: Service
          name: httpbin
---
apiVersion: networking.x.k8s.io/v1alpha1
kind: TcpRoute
metadata:
  name: tcp
  # TODO use different namespace
  namespace: istio-system
spec: {}
