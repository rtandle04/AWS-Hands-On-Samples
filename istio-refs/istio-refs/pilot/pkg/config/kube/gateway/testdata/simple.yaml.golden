apiVersion: networking.istio.io/v1alpha3
kind: Gateway
metadata:
  creationTimestamp: null
  name: gateway-istio-autogenerated-k8s-gateway
  namespace: istio-system
spec:
  selector:
    istio: ingressgateway
  servers:
  - hosts:
    - my.domain.example
    port:
      name: http-80-gateway-gateway-istio-system
      number: 80
      protocol: http
---
apiVersion: networking.istio.io/v1alpha3
kind: VirtualService
metadata:
  creationTimestamp: null
  name: http-istio-autogenerated-k8s-gateway
  namespace: istio-system
spec:
  gateways:
  - gateway-istio-autogenerated-k8s-gateway
  hosts:
  - my.domain.example
  http:
  - headers:
      request:
        add:
          my-added-header: added-value
        remove:
        - my-removed-header
    match:
    - headers:
        my-header:
          exact: some-value
      uri:
        prefix: /get
    route:
    - destination:
        host: httpbin.istio-system.svc.cluster.local
---
